hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AbfsConfiguration.java

```diff
@@ -517,6 +527,14 @@ public int getWriteBufferSize() {
     return this.writeBufferSize;
   }
 
+  public boolean readSmallFilesCompletely() {
```

Comment:
I recall it was discussed that small file criteria will be capped at 4Mb file size. I dont see that in the review. Was it scratched back to small file being one less than buffer size ?

Reply:
It was finalised as the criteria to consider a file small as the size smaller than the buffer size.

Reply:
Yes, if 100 MB is set as buffer size, current logic is forcing a single server read request for 100 MB.